proposals/low-level-struct-improvements.md

```diff
@@ -1024,3 +1047,51 @@ ref struct StackLinkedListNode<T>
     }
 }
 ```
+
+### Important Cases
+There are several important cases that need to be handled correctly by the rules. These are those cases and explanation of how the rules prevent them from happening.
+
+#### Preventing tricky ref assignment from readonly mutation
+When a `ref` is taken to a `readonly` field in a constructor or `init` member the type is `ref` not `ref readonly`. This is a long standing behavior that allows for code like the following:
+
+```c#
+struct S
+{
+    readonly int i; 
+
+    public S(string s)
+    {
+        M(ref i);
+    }
+
+    static void M(ref int i) { }
+}
+```
+
+That does pose a potential problem though if such a `ref` were able to be stored into a `ref` field on the same type. It would allow for direct mutation of a `readonly struct` from an instance member:
+
+```c#
+readonly ref struct S
+{ 
+    readonly int i; 
+    readonly ref r; 
```

Comment:
I'm a little late, but I assume this is meant to be `readonly ref int r;`?

Reply:
No this was meant to be `readonly ref`. This is meant to demonstrate a particularly nefarious example of how `ref` and `readonly` interact in ctors and the challenges with making sure we maintain the semantics of `readonly struct` after the ctor completes.

Reply:
I'm pretty sure I get the point of the example. I'm not disputing the `readonly ref` part, I'm disputing the lack of `int`.

Reply:
Oh ha. I misread you're original comment as `readonly ref in`. I thought you were using it as a `in` / aka readonly. Yep this is  miss and I need to add it.